sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -672,7 +673,7 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
             List<Task> tasks = new List<Task>(_maxConcurrentCalls + _receiverManagers.Count);
             try
             {
-                while (!cancellationToken.IsCancellationRequested)
+                while ((!cancellationToken.IsCancellationRequested) && (!Connection.IsClosed))
```

Comment:
nit:
```suggestion
                while (!cancellationToken.IsCancellationRequested && !Connection.IsClosed)
```

Reply:
I'll acquiesce because this is Service Bus, but know that it makes me sad.